game/scripts/vscripts/abilities/oaa_arcane_orb.lua

```diff
@@ -31,175 +43,209 @@ function modifier_oaa_arcane_orb:RemoveOnDeath()
   return false
 end
 
-function modifier_oaa_arcane_orb:OnCreated()
-  if IsServer() then
-    if not self.procRecords then
-      self.procRecords = {}
-    end
-    self.parentOriginalProjectile = self:GetParent():GetRangedProjectileName()
-    Debug.EnabledModules["abilities:oaa_arcane_orb"] = false
-  end
-end
-
-modifier_oaa_arcane_orb.OnRefresh = modifier_oaa_arcane_orb.OnCreated
-
 function modifier_oaa_arcane_orb:DeclareFunctions()
   return {
     MODIFIER_EVENT_ON_ATTACK_START,
     MODIFIER_EVENT_ON_ATTACK,
     MODIFIER_EVENT_ON_ATTACK_LANDED,
-    MODIFIER_EVENT_ON_ATTACK_FAIL
   }
 end
 
-function modifier_oaa_arcane_orb:OnAttackStart(keys)
+function modifier_oaa_arcane_orb:OnAttack(event)
   local parent = self:GetParent()
+  local ability = self:GetAbility()
 
-  if keys.attacker ~= parent then
+  if event.attacker ~= parent then
     return
   end
 
-  local ability = self:GetAbility()
-  local target = keys.target
-  -- Wrap in function to defer evaluation
-  local function autocast()
-    return (
-      target.GetUnitName and -- Check for existence of GetUnitName method to determine if target is a unit
```

Comment:
This check is here for a reason. All this stuff runs when you're attacking items too, which shouldn't use Arcane Orb attacks.

Reply:
I forgot about attacking items. I will add it.